### ISSUE TITLE
Warn on invalid use of tuples in `Literal`

### ISSUE THREAD

**Issue by Zac-HD** [python/mypy#11210]
[PEP-586 describes a variety of illegal parameters for `Literal`](https://www.python.org/dev/peps/pep-0586/#illegal-parameters-for-literal-at-type-check-time); use of which should cause typechecking to fail.

A similar check has been implemented in `pyright` (or PyLance?), where the following example emits `Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value`.  The examples I've seen in the wild tend to be `Literal[tuple(KNOWN_KEYS)]` or similar.

```python
from typing import Literal

zero: Literal[int()] = 0  # Should fail, not a literal
imag: Literal[1j] = 1j    # Should fail, float or complex not allowed
t: Literal[(3, 4)] = 3    # Should fail, could suggest `Tuple[Literal[3], Literal[4]]`
```

**Comment by hauntsaninja** [comment]
The first two examples actually do give me an error, it's only the third that doesn't. Seems to get interpreted the same as `Literal[3, 4]`, maybe a subscript slice bug?

**Comment by Zac-HD** [comment]
Ah, looks like our config included 

```ini
# Disable the error we get from Callable[SomeParamSpec, R]
# This will be fixed in the next version of Mypy (after 0.910)
disable_error_code = misc
```

which explains it.  I might go open an issue on the PEP repo instead for the last item; since you're correct that the runtime effect is the same we should probably get this clarified or removed - both `mypy` and `pyright` also complain about `Literal[3, (4, 5)]`, though it doesn't have the same ambiguity.
